check_api/src/main/java/com/google/errorprone/fixes/SuggestedFixes.java

```diff
@@ -1309,8 +1309,9 @@ public CharSequence getCharContent(boolean ignoreEncodingErrors) throws IOExcept
     boolean warningInSameCompilationUnit = false;
     for (Diagnostic<? extends JavaFileObject> diagnostic : diagnosticListener.getDiagnostics()) {
       warningIsError |= diagnostic.getCode().equals("compiler.err.warnings.and.werror");
-      boolean diagnosticInSameCompilationUnit =
-          diagnostic.getSource().toUri().equals(modifiedFileUri);
+      JavaFileObject diagnosticSource = diagnostic.getSource();
+      boolean diagnosticInSameCompilationUnit = diagnosticSource == null
+              || diagnosticSource.toUri().equals(modifiedFileUri);
```

Comment:
As the discussion in #1873 shows, different readers will have different expectations about whether it should be `diagnosticSource == null` or `diagnosticSource != null`. Perhaps it makes sense to add a comment with the reasoning from https://github.com/google/error-prone/issues/1873#issuecomment-755782269. Something like (modulo formatting):
```suggestion
      // If the source's origin is unknown, assume that new diagnostics are due to a modification.
      boolean diagnosticInSameCompilationUnit = diagnosticSource == null
              || diagnosticSource.toUri().equals(modifiedFileUri);
```